client/state/ui/guided-tours/selectors.js

```diff
@@ -30,10 +30,22 @@ export const getGuidedTourState = createSelector(
 		const tourState = getRawGuidedTourState( state );
 		const { shouldReallyShow, stepName = '' } = tourState;
 		const stepConfig = getToursConfig()[ stepName ] || false;
+
+		const shouldShow = (
+			// the Preview step requires a selected site
+			!! getSelectedSiteId( state ) &&
```

Comment:
Also, this makes the tour not appear if starting from e.g. Reader. I.e. http://calypso.localhost:3000/?tour=main results in no tour step sliding in. Removing it makes it work again, and the preview step is positioned just fine. 

Ideally I guess each step would have some sort of preconditions that are checked before displaying the step. If they're not met ... we jump to the step after that? Sounds a bit like halting problem though.